pkg/apis/pipeline/v1alpha1/pull_request_resource.go

```diff
@@ -0,0 +1,159 @@
+	}
+	prResource := PullRequestResource{
+		Name:    r.Name,
+		Type:    r.Spec.Type,
+		Secrets: r.Spec.SecretParams,
+	}
+	for _, param := range r.Spec.Params {
+		switch {
+		case strings.EqualFold(param.Name, "URL"):
+			prResource.URL = param.Value
+		}
+	}
+
+	return &prResource, nil
+}
+
+// GetName returns the name of the resource
+func (s PullRequestResource) GetName() string {
+	return s.Name
+}
+
+// GetType returns the type of the resource, in this case "Git"
+func (s PullRequestResource) GetType() PipelineResourceType {
+	return PipelineResourceTypePullRequest
+}
+
+// GetURL returns the url to be used with this resource
+func (s *PullRequestResource) GetURL() string {
+	return s.URL
+}
+
+// GetParams returns the resource params
+func (s PullRequestResource) GetParams() []Param { return []Param{} }
+
+// Replacements is used for template replacement on a PullRequestResource inside of a Taskrun.
+func (s *PullRequestResource) Replacements() map[string]string {
+	return map[string]string{
+		"name": s.Name,
+		"type": string(s.Type),
+		"url":  s.URL,
+	}
+}
+
+func (s *PullRequestResource) GetDownloadContainerSpec() ([]corev1.Container, error) {
+	args := []string{"-url", s.URL, "-path", s.Name, "-mode", "download"}
+
+	evs := []corev1.EnvVar{}
+	for _, sec := range s.Secrets {
+		switch {
+		case strings.EqualFold(sec.FieldName, "githubOauthToken"):
+			fmt.Println("ADDING ENV VAR")
+			ev := corev1.EnvVar{
+				Name: strings.ToUpper(sec.FieldName),
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: sec.SecretName,
+						},
+						Key: sec.SecretKey,
+					},
+				},
+			}
+			evs = append(evs, ev)
+		}
+	}
+
+	return []corev1.Container{{
+		Name:       names.SimpleNameGenerator.RestrictLengthWithRandomSuffix(prSource + "-" + s.Name),
+		Image:      *prImage,
+		Command:    []string{"/ko-app/pullrequest-init"},
+		Args:       args,
+		WorkingDir: workspaceDir,
+		Env:        evs,
+	}}, nil
+}
+
+func (s *PullRequestResource) GetUploadContainerSpec() ([]corev1.Container, error) {
+	args := []string{"-url", s.URL, "-path", s.Name, "-mode", "upload"}
+
+	evs := []corev1.EnvVar{}
+	for _, sec := range s.Secrets {
+		switch {
+		case strings.EqualFold(sec.FieldName, "githubOauthToken"):
+			ev := corev1.EnvVar{
+				Name: strings.ToUpper(sec.FieldName),
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: sec.SecretName,
+						},
+						Key: sec.SecretKey,
+					},
+				},
+			}
+			evs = append(evs, ev)
+		}
+	}
+
+	return []corev1.Container{{
+		Name:       names.SimpleNameGenerator.RestrictLengthWithRandomSuffix(prSource + "-" + s.Name),
+		Image:      *prImage,
+		Command:    []string{"/ko-app/pullrequest-init"},
+		Args:       args,
+		WorkingDir: workspaceDir,
+		Env:        evs,
+	}}, nil
+}
```

Comment:
could you include unit tests for these as well? (e.g. like https://github.com/tektoncd/pipeline/blob/4aac2b6cb197a65aa8c93595cceeb417bfd7b59e/pkg/apis/pipeline/v1alpha1/gcs_resource_test.go)

Reply:
Done.